packages/headless/src/features/commerce/pagination/pagination-slice.ts

```diff
@@ -22,12 +23,13 @@ export const paginationReducer = createReducer(
           state.page = action.payload;
         }
       })
-      .addCase(fetchProductListing.fulfilled, (state, action) => {
-        const responsePagination = action.payload.response.pagination;
-        state.perPage = responsePagination.perPage;
-        state.totalCount = responsePagination.totalCount;
-        state.totalPages = responsePagination.totalPages;
-        // TODO: replace with state = responsePagination once the API response pagination returns the current page.
-      });
+      .addCase(
+        fetchProductListing.fulfilled,
+        (_, action) => action.payload.response.pagination
```

Comment:
Please forgive me my noobness with redux, but how does this knows the correct context when both a search and listing pagination is on the same page?

Reply:
It doesn't! :) This is why we'll eventually want to adopt one of the propositions in [this RFC](https://docs.google.com/document/d/1shl4QnnDnQVZat00QT4PqcLAsvKUbGo_zTqwX3iZJWg/edit#heading=h.jgfle3phkvb2), which will allow us to have multiple commerce solution types active on the same page.